### PR TITLE
Several changes to SMS notifications:

### PR DESCRIPTION
**Pull request by lisadu**
* Send 3 notifications before requiring re-subscribe
* Include nearby zips
* Use firebase db filter to only get selected zips, potentially avoids pagination issues with trying to retrieve all zips (this part hasn't been tested yet, I need to get Firebase)

### CODE REVIEW

File: helpers/zip_helpers.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
         if is_valid_zip(match):
             return match
+
+DEFAULT_NEARBY_COORD = 0.005  # Roughly 5 miles
+def get_closest_zips(zip, distance=DEFAULT_NEARBY_COORD):
+    """Returns a list of nearby zips, including the input zip"""
+    long, lat = ALL_ZIPS[zip]
+    zips = []
+    for zip2, coord in ALL_ZIPS.items():
+        if (abs(coord[0]-long)**2 + abs(coord[1]-lat)**2 < distance):
```
Comment (likhter) [inline helpers/zip_helpers.py]: sqrt or distance ** 2 ?

File: helpers/database.py [hunk 2]
```diff
@@ -17,13 +19,12 @@ def get_where(self, **kwargs):
         """Returns a user that satisfies condition given by kwarg (e.g. zip_code='12345')"""
         assert len(kwargs) == 1
         match_k, match_v = list(kwargs.items())[0]
-        for key, user in self.get().items():
-            user['id'] = key
-            if user[match_k] == match_v:
-                return user
+        query = {'orderBy': match_k, 'equalTo': match_v}
+        return self.firebase.get('users', None, params=query.update(self.params)) or {}
```
Comment (mazore) [inline helpers/database.py]: `dict.update` returns None, needs to be changed to
```python
query = {'orderBy': match_k, 'equalTo': match_v}
query.update(self.params)
return self.firebase.get('users', None, params=query) or {}
```

**Review by mazore — CHANGES_REQUESTED** [review]

File: helpers/database.py [hunk 3]
```diff
@@ -17,13 +19,12 @@ def get_where(self, **kwargs):
         """Returns a user that satisfies condition given by kwarg (e.g. zip_code='12345')"""
         assert len(kwargs) == 1
         match_k, match_v = list(kwargs.items())[0]
-        for key, user in self.get().items():
-            user['id'] = key
-            if user[match_k] == match_v:
-                return user
+        query = {'orderBy': match_k, 'equalTo': match_v}
```
Comment (mazore) [inline helpers/database.py]: Parameters need to be enclosed in double quotes

File: helpers/database.py [hunk 4]
```diff
@@ -19,8 +19,9 @@ def get_where(self, **kwargs):
         """Returns a user that satisfies condition given by kwarg (e.g. zip_code='12345')"""
         assert len(kwargs) == 1
         match_k, match_v = list(kwargs.items())[0]
-        query = {'orderBy': match_k, 'equalTo': match_v}
-        return self.firebase.get('users', None, params=query.update(self.params)) or {}
+        query = {'orderBy': f'"{match_k}"', 'equalTo': f'"{match_v}"'}
```
Comment (lisadu) [inline helpers/database.py:24]: @mazore are you sure the query params need to be quoted like that? Are you able to test this with firebase? I don't have access to the account.

**Comment by lisadu** [comment]
@mazore Please merge this and deploy when you get a chance

File: helpers/database.py (hunk 4 above)
Comment (mazore) [reply helpers/database.py:24]: Yes I’ve tested it out and this is how it works

**Comment by mazore** [comment]
I’m going to do some more test to make sure it’s ok, then I’ll merge

**Pull request merged**